numba/stencilparfor.py

```diff
@@ -674,3 +680,31 @@ def __init__(self, typingctx, targetctx, args, f_ir):
         self.typemap = None
         self.return_type = None
         self.calltypes = None
+
+
+def _get_const_index_expr(kernel_defs, func_ir, index_var):
+    """
+    infer index_var as constant if it is of the form -c where c is a constant
+    in the outer function. index_var is assumed to be inside stencil kernel
+    """
+    try:
+        require(isinstance(index_var, ir.Var))
+        # case where the index is a const itself in outer function
+        var_const =  guard(ir_utils.find_const, func_ir, index_var)
+        if var_const is not None:
+            return var_const
+        # match definition inner_var = unary(index_var)
+        var_def_list = kernel_defs[index_var.name]
+        require(len(var_def_list) == 1)
+        var_def = var_def_list[0]
+        require(isinstance(var_def, ir.Expr) and var_def.op == 'unary')
```

Comment:
This line isn't obvious to me.  If the kernel was specified something like "a[c-2]" with c=1 in the outer scope, then what is going to happen here?  Ideally, anything resolvable at compile time we could handle but until then falling back and saying that we couldn't compute the kernel bounds statically is fine.  We just shouldn't generate an assertion that this requirement on this line isn't true.

Reply:
This is only handling the `-c` case and goes back to default behavior if not possible. This `require` is inside try/except so it won't raise error to the user.

Looking again, I think handling binary expression is not too hard and will enable significant cases. I'll add support for it.

Reply:
done.